service/history/replication/executable_history_task.go

```diff
@@ -297,3 +302,145 @@ func (e *ExecutableHistoryTask) getDeserializedEvents() (_ []*historypb.HistoryE
 	}
 	return events, newRunEvents, err
 }
+
+func (e *ExecutableHistoryTask) BatchWith(incomingTask BatchableTask) (_ TrackableExecutableTask, retErr error) {
```

Comment:
`func (e *ExecutableHistoryTask) BatchWith(incomingTask BatchableTask) (TrackableExecutableTask, bool) {`?

return a bool indicating whether batch is successful is more appropriate

Reply:
i see, validation is also done internally .....

Reply:
Sure. Will change.